dokka-subprojects/plugin-base/src/main/kotlin/org/jetbrains/dokka/base/transformers/documentables/JvmMappedMethodsDocumentableFilterTransformer.kt

```diff
@@ -0,0 +1,258 @@
+                    "indexOf(Ljava/lang/String;)I",
+                    "indexOf(Ljava/lang/String;I)I",
+                    "intern()Ljava/lang/String;",
+                    "isEmpty()Z",
+                    "lastIndexOf(I)I",
+                    "lastIndexOf(II)I",
+                    "lastIndexOf(Ljava/lang/String;)I",
+                    "lastIndexOf(Ljava/lang/String;I)I",
+                    "matches(Ljava/lang/String;)Z",
+                    "offsetByCodePoints(II)I",
+                    "regionMatches(ILjava/lang/String;II)Z",
+                    "regionMatches(ZILjava/lang/String;II)Z",
+                    "replaceAll(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;",
+                    "replace(CC)Ljava/lang/String;",
+                    "replaceFirst(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;",
+                    "replace(Ljava/lang/CharSequence;Ljava/lang/CharSequence;)Ljava/lang/String;",
+                    "split(Ljava/lang/String;I)[Ljava/lang/String;",
+                    "split(Ljava/lang/String;)[Ljava/lang/String;",
+                    "startsWith(Ljava/lang/String;I)Z",
+                    "startsWith(Ljava/lang/String;)Z",
+                    "substring(II)Ljava/lang/String;",
+                    "substring(I)Ljava/lang/String;",
+                    "toCharArray()[C",
+                    "toLowerCase()Ljava/lang/String;",
+                    "toLowerCase(Ljava/util/Locale;)Ljava/lang/String;",
+                    "toUpperCase()Ljava/lang/String;",
+                    "toUpperCase(Ljava/util/Locale;)Ljava/lang/String;",
+                    "trim()Ljava/lang/String;",
+                    "isBlank()Z",
+                    "lines()Ljava/util/stream/Stream;",
+                    "repeat(I)Ljava/lang/String;"
+                ) +
+
+                inJavaLang("Double", "isInfinite()Z", "isNaN()Z") + inJavaLang("Float", "isInfinite()Z", "isNaN()Z") +
+
+                inJavaLang("Enum", "getDeclaringClass()Ljava/lang/Class;", "finalize()V") + inJavaLang(
+            "CharSequence",
+            "isEmpty()Z"
+        )
+
+
+    private val deprecationAnnotationDRI = DRI("kotlin", "Deprecated")
+
+    /**
+     * KT-65438: In K1, such methods (e.g. translateEscapes, formatted... in java.lang.String) have the NOT_CONSIDERED status,
+     * that means they are available and have a deprecation annotation.
+     * In K2, they are unavailable for final classes + [NOT_CONSIDER_METHOD_SIGNATURES]
+     * @see NOT_CONSIDER_METHOD_SIGNATURES
+    */
+    private fun DFunction.hasDeprecated(): Boolean = this.extra[Annotations]?.directAnnotations?.any {
+        it.value.any {
+            it.dri == deprecationAnnotationDRI && it.params["message"] == StringValue("This member is not fully supported by Kotlin compiler, so it may be absent or have different signature in next major version")
+        }
+    } ?: false
+
+    override fun shouldBeSuppressed(d: Documentable): Boolean =
```

Comment:
could you please rewrite it to an ordinary function without using `let`, but using early returns?
It is hard to understand what is the full condition for suppression is